src/Magento/FunctionalTestingFramework/Test/Objects/ActionObject.php

```diff
@@ -293,6 +295,54 @@ public function resolveReferences()
         }
     }
 
+    /**
+     * @throws TestReferenceException
+     * @throws XmlException
+     */
+    private function resolveHelperReferences()
+    {
+        if ($this->getType() !== 'helper') {
+            return;
+        }
+        $isResolved = false;
+
+        try {
+            foreach ($this->actionAttributes as $attrKey => $attrValue) {
+                $this->actionAttributes[$attrKey] = $this->findAndReplaceReferences(SectionObjectHandler::getInstance(), $attrValue);
+            }
+            $isResolved = true;
+        } catch (\Exception $e) {
+            // catching exception to allow other entity type resolution to proceed
+        }
+
+        try {
+            foreach ($this->actionAttributes as $attrKey => $attrValue) {
+                $this->actionAttributes[$attrKey] = $this->findAndReplaceReferences(PageObjectHandler::getInstance(), $attrValue);
+            }
+            $isResolved = true;
+        } catch (\Exception $e) {
+            // catching exception to allow other entity type resolution to proceed
+        }
+
+        try {
+            foreach ($this->actionAttributes as $attrKey => $attrValue) {
+                $this->actionAttributes[$attrKey] = $this->findAndReplaceReferences(DataObjectHandler::getInstance(), $attrValue);
+            }
+            $isResolved = true;
+        } catch (\Exception $e) {
+            // catching exception to allow other entity type resolution to proceed
+        }
+
+        if ($isResolved !== true) {
```

Comment:
What if there is nothing to resolve? Would this still work?

Reply:
Test it, please 😄

Reply:
but short answer is yes, it will work.

Reply:
tested it. it worked 👍